### ISSUE TITLE
Doubling of titlebar

### ISSUE THREAD

**Issue by vitaly-zdanevich** [nulloy/nulloy#177]
Sway (Wayland)

![20220112_23h50m50s_grim](https://user-images.githubusercontent.com/3514015/149219745-7a3fff13-1c2c-4beb-aa70-574d881be8b2.png)

**Comment by noscript** [comment]
Does Sway even support undecorated windows?

You can get rid of the border if you patch the skin. Open any `.nzs` skin file with archive manager, edit `script.js` and append `Ui.titleWidget.setVisible(false);` to the end of `Main()` function.

**Comment by vitaly-zdanevich** [comment]
Did it with `/usr/share/nulloy/skins/slim.nzs::script.js`, start Nulloy - but still have two title bars.

**Comment by vitaly-zdanevich** [comment]
> Does Sway even support undecorated windows?

I see only Nulloy title bar if go to the full screen by <kbd>$mod</kbd>+<kbd>f</kbd>

**Comment by noscript** [comment]
> Did it with `/usr/share/nulloy/skins/slim.nzs::script.js`, start Nulloy - but still have two title bars.

Did you remember to update `slim.nzs` with the patched `script.js`?

**Comment by vitaly-zdanevich** [comment]
Yes, I edited it through Vim, in place.

**Comment by noscript** [comment]
Please attach the `.nzs` file, I will take a look.

**Comment by vitaly-zdanevich** [comment]
[slim.zip](https://github.com/nulloy/nulloy/files/7870160/slim.zip)

Renamed to zip because I cannot upload nzs here.

**Comment by vitaly-zdanevich** [comment]
Switched to i3 - here I do not see the title bar of Nulloy.

**Comment by noscript** [comment]
> [slim.zip](https://github.com/nulloy/nulloy/files/7870160/slim.zip)

The change is correct.

> Switched to i3 - here I do not see the title bar of Nulloy.

It shouldn't matter if it's i3 or something else.

Do you run the player in both cases in exactly the same way? If you run `./nulloy` without installing, then it will load skins from `./skins/` and not from `/usr/share/nulloy/skins/`.

**Comment by vitaly-zdanevich** [comment]
> Do you run the player in both cases in exactly the same way?

Yes, from hotkey, defined in respective config files, just calling `nulloy` from PATH.

**Comment by noscript** [comment]
I have no idea why the change would be picked up in i3 and not sway. Could please you try removing `/usr/share/nulloy/skins/slim.nzs` and see if the skin is still loaded.

**Comment by vitaly-zdanevich** [comment]
I removed Wayland related stuff, I am afraid to mess my Gentoo :(

**Comment by noscript** [comment]
So there is no more problem with double titlebar?

**Comment by vitaly-zdanevich** [comment]
Adding line of JS is not a good solution, better to add such option to preferences.

**Comment by noscript** [comment]
I would need to know how `wmctrl` names i3. Please run `wmctrl -m` and paste the output.

**Comment by vitaly-zdanevich** [comment]
`wmctrl: command not found`

**Comment by noscript** [comment]
Please install it.

**Comment by vitaly-zdanevich** [comment]
```
# wmctrl -m
Name: i3
Class: N/A
PID: N/A
Window manager's "showing the desktop" mode: N/A
```
